cmd/faucet/faucet.go

```diff
@@ -745,17 +761,30 @@ func (f *faucet) loop() {
 			f.lock.RUnlock()
 		}
 	}()
+
 	// Wait for various events and assing to the appropriate background threads
+	subscribeTimeoutTimer := time.NewTimer(subscribeTimeout)
+
 	for {
 		select {
 		case head := <-heads:
+			if !subscribeTimeoutTimer.Stop() {
```

Comment:
No need to reset timer twice, since we'll write data to `update` chan, which will reset timer in only one place.

Reply:
ok, remove it [53132d8](https://github.com/antimatter-dao/bas-template-bsc/pull/18/commits/53132d8e1d5b002bb15d89f062ffdf2857501692)